src/bidderSettings.js

```diff
@@ -0,0 +1,67 @@
+import {deepAccess, mergeDeep} from './utils.js';
+import {getGlobal} from './prebidGlobal.js';
+
+const CONSTANTS = require('./constants.json');
+
+export function ScopedSettings(getSettings, defaultScope) {
```

Comment:
Is there a reason to not using Class if you choose OOP?

Reply:
> Is there a reason to not using Class if you choose OOP?

Because when I started working on this IE11 was still a target :) I don't have a strong preference on the syntax, I can rewrite using `class` - should I take your comment as an encouragement?

> I'm not big fan of OOP concept.

Then some day I'll have to explain to you why you're wrong.

But more seriously, I don't understand how you can say we barely use OOP - config, auctionManager, auction, adapterManager, adapters, etc are all objects with capabilities. Is the problem that I use `prototype` instead of hidden closures as "private" methods? or the fact that you have to use `new` for this function but not for the others?

Reply:
>  should I take your comment as an encouragement?

Yes, totally :)